journalpump/journalpump.py

```diff
@@ -675,6 +718,32 @@ def _truncate_long_message(self, json_entry):
         }
         return json.dumps(entry, default=default_json_serialization).encode("utf8")
 
+    def _apply_secret_filters(self, data):
+        changed = False
+
+        for secret_filter in self.reader.secret_filters:
+            msg = data.get("MESSAGE")
+            if msg:
+                match = secret_filter["compiled_pattern"].match(msg)
+                if match:
+                    msg = ""
+                    changed = True
+
+                    self.pump.stats.increase(
```

Comment:
converted to draft because i forgot to put this metric emission behind a config option as requested.